lib/irb.rb

```diff
@@ -1222,6 +1222,9 @@ def handle_exception(exc)
         irb_bug = true
       else
         irb_bug = false
+        backtrace = exc.backtrace.map { |l| @context.workspace.filter_backtrace(l) }.compact
+        exc = exc.clone
+        exc.set_backtrace(backtrace)
```

Comment:
I think we can add some comments to make the intention and context clearer:

```suggestion
        # This is mostly to make IRB work nicely with Rails console's backtrace filtering, which patches WorkSpace#filter_backtrace
        # In such use case, we want to filter the exception's backtrace before its displayed through Exception#full_message
        # And we clone the exception object in order to avoid mutating the original exception
        # TODO: introduce better API to expose exception backtrace externally
        backtrace = exc.backtrace.map { |l| @context.workspace.filter_backtrace(l) }.compact
        exc = exc.clone
        exc.set_backtrace(backtrace)
```

Reply:
Thanks Stan!

What will release for this look like? Will it be patched onto older versions of irb or available on the next release? I have no idea how these things work. I'm wondering when we can mark the rails issue as fixed

Reply:
This will be included in the next IRB release (1.13 or 1.12.1). If Rails apps bump their IRB version, they'll be running Rails console with this patch.

Reply:
Awesome, thanks!